packages/perspective-viewer-d3fc/src/js/charts/sunburst.js

```diff
@@ -68,23 +70,23 @@ function sunburst(container, settings) {
             title.attr("transform", `translate(0, ${-(height / 2 - 5)})`);
 
             const radius = (Math.min(width, height) - 120) / 6;
-            const color = treeColor(settings, split, data.data.children);
             sunburstSeries()
                 .settings(settings)
                 .split(split)
                 .data(data)
                 .color(color)
                 .radius(radius)(sunburstElement);
 
-            if (color) {
-                const legend = colorRangeLegend().scale(color);
-                select(svgNode.parentNode)
-                    .call(legend)
-                    .select("div.legend-container");
-            }
-
             tooltip().settings(settings)(sunburstElement.selectAll("g.segment"));
         });
+
+    if (color) {
+        const legend = colorRangeLegend().scale(color);
+        container
+            .call(legend)
+            .select("div.legend-container")
+            .style("transform", `translateY(${containerTop}px)`);
```

Comment:
Unfortunately this doesn't work if there are more than one `perspective-viewer` on the page. The `fixed` position with offset gives it an offset from the top of the page, rather than from the top of the component.

I think we could do this by putting an extra `<div>` inside the container. The grid (with charts) would go inside the `<div>`, which would be scrollable. The legend would be next to it in the non-scrollable container.

Reply:
OK, I'll have a look at that

Reply:
Did you have a specific example? It seems to work for me:

![image](https://user-images.githubusercontent.com/29425751/55634336-323c7c80-57b6-11e9-958e-d2709bb2b860.png)

Reply:
Actually, it doesn't work if the whole page can scroll.

Reply:
My apologies. It looks like I broke it myself.

The whole-page-scrolling problem would still be a problem. I'd typically avoid using `fixed` for just this sort of reason - especially if you have to re-position it by measuring the element it's supposed to appear to be in.

Reply:
To elaborate on my previous suggestion - you'd probably want to make the "grid" `div` absolute position at 100% width and height (and scrollable), but with a right-padding to allow for the legend. That way, the scroll bar should still appear in the same place.

Just as a further thought. The treemap is likely to want a similar layout, so perhaps the grid-layout-with-legend logic could be extracted into a reusable component?

Reply:
I'm having a look into it now. Nothing is displaying in the inner div at the moment but it shouldn't be too hard once I figure that bit out.

Reply:
Right, I now have it working with absolute position, I need to figure out how to extract this.